packages/driver/src/cy/location.ts

```diff
@@ -20,6 +33,29 @@ export const create = (state: StateFunc) => ({
       return ''
     }
   },
+  async getCrossOriginRemoteLocation (key?: string | undefined, win?: Window) {
```

Comment:
I spent some time trying to think about how we can convey what is going on here. Since I don't think we can always know the value of `key` statically all the time, I wonder if it's best to return an `unknown` type and assume casting in the caller, with the idea that you might need to handle errors there. No idea if that was comprehensible or not 😄 . For instance: 

```

try {
  const url = await cy.getCrossOriginRemoteLocation('href') as string
} catch(e){
  // probably made a bad assumption?
}
```
7
```suggestion
  async getCrossOriginRemoteLocation (key?: keyof LocationObject , win?: Window): Promise<unknown> {
```

Reply:
Also wonder if it just makes sense to make this function signature simpler. Either return `undefined` or a `LocationObject`.

Reply:
I created this function based on the api of (and as a 'drop in' replacement for) `getRemoteLocation`.

Personally i think this function is doing too much and should either be split into two functions one that returns a location object and one that picks keys off it or just return a locaiton object and let users pick they key they want, not like it's hard to do that.

Regardless, I'd like to change this in a separate pr, if we do it at all.